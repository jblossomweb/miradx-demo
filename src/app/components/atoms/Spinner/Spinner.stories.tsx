import React from 'react'
import type { Meta, StoryObj } from '@storybook/react'

import Spinner from './Spinner'

type Story = StoryObj<typeof meta>

const meta = {
  title: 'components/atoms/Spinner',
  component: Spinner,
  render: (args) => <Spinner {...args} />,
} satisfies Meta<typeof Spinner>

export const Sample: Story = {
  args: {},
}

export const Sized: Story = {
  args: {
    size: 80,
  },
}

export default meta
