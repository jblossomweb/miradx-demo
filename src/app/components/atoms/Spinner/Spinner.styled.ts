import tw from 'tailwind-styled-components'
import { BiLoaderAlt as LoaderAlt } from 'react-icons/bi'

export interface SizeProps {
  $size: number
}

export const Spinner = tw(LoaderAlt)`
  animate-spinfast
`
