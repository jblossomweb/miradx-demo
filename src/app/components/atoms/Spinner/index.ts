export { default, type SpinnerProps } from './Spinner'
