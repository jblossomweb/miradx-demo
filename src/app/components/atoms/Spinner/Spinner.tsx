import React from 'react'
import * as Styled from './Spinner.styled'

export interface SpinnerProps {
  size?: number
}

const Spinner: React.FC<SpinnerProps> = ({ size = 40 }) => (
  <Styled.Spinner
    size={size}
    role="status"
  />
)

export default Spinner
