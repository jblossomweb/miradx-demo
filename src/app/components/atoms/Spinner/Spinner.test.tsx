import React from 'react'
import { render, screen } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import Spinner from './Spinner'
import * as stories from './Spinner.stories'

describe('app/components/atoms/Spinner', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('assertions', () => {
    const renderDefault = () => render(<Spinner />)

    beforeEach(jest.clearAllMocks)

    it('renders a spinner with status role', () => {
      renderDefault()
      const spinner = screen.getByRole('status', {
        hidden: true,
      }) as HTMLButtonElement
      expect(spinner).toBeInTheDocument()
    })

    it('spins the spinner', () => {
      renderDefault()
      const spinner = screen.getByRole('status', {
        hidden: true,
      }) as HTMLButtonElement
      expect(spinner).toHaveClass('animate-spinfast')
    })
  })
})
