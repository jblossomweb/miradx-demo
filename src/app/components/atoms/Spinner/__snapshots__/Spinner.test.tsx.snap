// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Spinner stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <svg
    class="animate-spinfast"
    fill="currentColor"
    height="40"
    role="status"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 22c5.421 0 10-4.579 10-10h-2c0 4.337-3.663 8-8 8s-8-3.663-8-8c0-4.336 3.663-8 8-8V2C6.579 2 2 6.58 2 12c0 5.421 4.579 10 10 10z"
    />
  </svg>
</DocumentFragment>
`;

exports[`app/components/atoms/Spinner stories Sized matches markup snapshot 1`] = `
<DocumentFragment>
  <svg
    class="animate-spinfast"
    fill="currentColor"
    height="80"
    role="status"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="80"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 22c5.421 0 10-4.579 10-10h-2c0 4.337-3.663 8-8 8s-8-3.663-8-8c0-4.336 3.663-8 8-8V2C6.579 2 2 6.58 2 12c0 5.421 4.579 10 10 10z"
    />
  </svg>
</DocumentFragment>
`;
